hw/drivers/sensors/dps368/src/dps368_comm_interface.c

```diff
@@ -0,0 +1,325 @@
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/*
+ * dps368_priv.h
+ *
+ *  Created on: Apr 15, 2019
+ *      Author: Saumitra Chafekar
+ */
+#include "dps368/dps368.h"
+#include "dps368_priv.h"
+
+
+/* Define the stats section and records */
+STATS_SECT_START(dps368_stat_section)
+    STATS_SECT_ENTRY(read_errors)
+    STATS_SECT_ENTRY(write_errors)
+    STATS_SECT_ENTRY(out_of_bound_data_errors)
+STATS_SECT_END
+
+/* Define stat names for querying */
+STATS_NAME_START(dps368_stat_section)
+    STATS_NAME(dps368_stat_section, read_errors)
+    STATS_NAME(dps368_stat_section, write_errors)
+    STATS_NAME(dps368_stat_section, out_of_bound_data_errors)
+STATS_NAME_END(dps368_stat_section)
+
+/* Global variable used to hold stats data */
+STATS_SECT_DECL(dps368_stat_section) g_dps368_stats;
+
+
+/**
+ * sensor stats init
+ *
+ *
+ * @param ptr Device structure
+ *
+ * @return	nothing
+ */
+void
+dps368_stats_int(struct os_dev *dev)
+{
+	int rc;
+	/* Initialise the stats entry */
+	rc = stats_init(
+			STATS_HDR(g_dps368_stats),
+			STATS_SIZE_INIT_PARMS(g_dps368_stats, STATS_SIZE_32),
+			STATS_NAME_INIT_PARMS(dps368_stat_section));
+	SYSINIT_PANIC_ASSERT(rc == 0);
+	/* Register the entry with the stats registry */
+	rc = stats_register(dev->od_name, STATS_HDR(g_dps368_stats));
+	SYSINIT_PANIC_ASSERT(rc == 0);
+}
+
+
+/*DPS368 bus W/R handling */
+#if !MYNEWT_VAL(BUS_DRIVER_PRESENT)
+/**
+ * Writes a single byte to the specified register using i2c
+ * interface
+ *
+ * @param The sensor interface
+ * @param The register address to write to
+ * @param The value to write
+ *
+ * @return 0 on success, non-zero error on failure.
+ */
+static int
+dps368_i2c_write_reg(struct sensor_itf *itf, uint8_t reg, uint8_t value)
+{
+    int rc;
+    uint8_t payload[2] = { reg, value };
+
+    struct hal_i2c_master_data data_struct = {
+        .address = itf->si_addr,
+        .len = 2,
+        .buffer = payload
+    };
+
+    rc = i2cn_master_write(itf->si_num, &data_struct, MYNEWT_VAL(DPS368_I2C_TIMEOUT_TICKS), 1,
+                           MYNEWT_VAL(DPS368_I2C_RETRIES));
+
+    if (rc) {
+    	DPS368_LOG(ERROR,
+                    "Could not write to 0x%02X:0x%02X with value 0x%02X\n",
+                    itf->si_addr, reg, value);
+        STATS_INC(g_dps368_stats, read_errors);
+    }
+
+    return rc;
+}
+
+/**
+ * Writes a single byte to the specified register using SPI
+ * interface
+ *
+ * @param The sensor interface
+ * @param The register address to write to
+ * @param The value to write
+ *
+ * @return 0 on success, non-zero error on failure.
+ */
+static int
+dps368_spi_write_reg(struct sensor_itf *itf, uint8_t reg, uint8_t value)

```

Comment:
This function should be wrapped with a #IF that checks if SPI is present. This file currently fails to build on systems without SPI.